mozilla-release/browser/components/preferences/in-content/preferences.js

```diff
@@ -58,11 +58,15 @@ function init_all() {
   register_module("paneSecurity", gSecurityPane);
   register_module("paneConnect", gConnectPane);
   // CLIQZ: DB-1230: Display the rich list item when connect module is available
-  fetch('chrome://cliqz/content/pairing/index.html').then( function(res) {
-    if(res.status === 200) {
-      document.getElementById('category-connect').hidden = false;
-    }
-  });
+  try {
+    fetch('chrome://cliqz/content/pairing/index.html').then( function(res) {
+      if(res.status === 200 && Services.prefs.getBoolPref('extensions.cliqz.connect')) {
```

Comment:
@mai-cliqz getBoolPref throws an exception if the pref is not there. I would suggest to change this if to : 

const CONNECT_PREF_NAME = 'extensions.cliqz.connect';
if(res.status === 200 && Services.prefs.getPrefType(CONNECT_PREF_NAME) == 128 /* bool */ && Services.prefs.getBoolPref(CONNECT_PREF_NAME))